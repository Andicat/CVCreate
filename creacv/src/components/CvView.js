import React from 'react';
import PropTypes from 'prop-types';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';

import CvBlock from './CvBlock';
import {createStyle} from './utils';

class CvView extends React.PureComponent {

    static propTypes = {
        blocks: PropTypes.array,
        stylePage: PropTypes.object,
    };

    state = {
        viewForPrint: false,
    }

    viewForPrint = () => {
        this.setState({viewForPrint:!this.state.viewForPrint});
    }
    
    render () {
        var cvBlocksCode = this.props.blocks.map( b => {
            return <CvBlock key={b.id} id={b.id} data={b} editable={false}></CvBlock>
        });

        return (
            <React.Fragment>
                {!this.state.viewForPrint && (
                    <header className={'header ' + this.props.transitionClass}>
                        <ul className='header__menu'>
                            <li className='header__menu-item'>
                                <button className='header__button header__button--print' onClick={this.viewForPrint}>View for print</button>
                            </li>
                            <li className='header__menu-item'>
                                <NavLink to='/cv' className='header__button header__button--edit'>Back to edit</NavLink>    
                            </li>
                        </ul>
                    </header>
                )}
                <main className={'main ' + this.props.transitionClass}>
                    {this.state.viewForPrint && (
                        <div className='header__print no-print'>
                            <span>You can save this page to PDF-file. Open a webpage in Chrome, press Ctrl+P to open the Print dialog and change the destination printer to “Save as PDF”. Hit the “Print” button and the webpage will download as a PDF document
                            </span>
                            <NavLink to='/cv' className='header__button header__button--edit'>Back to edit</NavLink>
                        </div>
                    )}
                    <div className='cv-view'>
                        <div className='cv' style={createStyle(this.props.stylePage)}>
                            {cvBlocksCode}
                        </div>
                    </div>
                </main>
            </React.Fragment>
        );
    }
}

const mapStateToProps = function (state) {
    return {
        blocks: state.cvData.blocks,
        stylePage: state.cvData.stylePage,
    };
};
  
export default connect(mapStateToProps)(CvView);
