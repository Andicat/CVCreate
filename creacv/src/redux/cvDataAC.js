const CV_BLOCK_ADD = 'CV_BLOCK_ADD';
const CV_BLOCK_DELETE = 'CV_BLOCK_DELETE';
const CV_BLOCK_MOVE = 'CV_BLOCK_MOVE';
const CV_BLOCK_ACTIVATE = 'CV_BLOCK_ACTIVATE';

const cvBlock_add = function(block) {
  return {
    type: CV_BLOCK_ADD,
    block:block,
  };
}

const cvBlock_delete = function(block) {
  return {
    type: CV_BLOCK_ADD,
    block:block,
  };
}

const cvBlock_move = function(block) {
  return {
    type: CV_BLOCK_MOVE,
    block:block,
  };
}

const cvBlock_activate = function(block) {
  return {
    type: CV_BLOCK_ACTIVATE,
    block:block,
  };
}

export {
  cvBlock_add, CV_BLOCK_ADD,
  cvBlock_delete, CV_BLOCK_DELETE,
  cvBlock_move, CV_BLOCK_MOVE,
  cvBlock_activate, CV_BLOCK_ACTIVATE,
}
