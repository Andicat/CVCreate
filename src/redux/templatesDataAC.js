const TEMPLATE_LOAD = 'TEMPLATE_LOAD';

const templates_load = function(data) {
    return {
        type: TEMPLATE_LOAD,
        data: data,
    };
}

export { templates_load, TEMPLATE_LOAD }
