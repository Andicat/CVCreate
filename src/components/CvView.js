import React from 'react';
import PropTypes from 'prop-types';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';
import Media from 'react-media';
import {cv_setLink} from '../redux/cvDataAC';
import CvBlock from './CvBlock';
import {createStyle, saveLocalStorage} from '../modules/utils';
import {saveFirebase} from './withDataLoad';

//просмотр документа
class CvView extends React.PureComponent {

    static propTypes = {
        blocks: PropTypes.array,
        stylePage: PropTypes.object,
        user: PropTypes.string,
        link: PropTypes.string,
    };

    state = {
        viewForPrint: false,
        linkIsUpdated: false,
    }

    //просмотр для печати
    viewForPrint = () => {
        this.setState({viewForPrint:!this.state.viewForPrint});
    }

    //создаем ссылку в базе приложения
    createLink = async () => {
        let linkName = this.props.user;
        let stateToSave = {style:this.props.stylePage,blocks:this.props.blocks};
        saveFirebase('Links',linkName,stateToSave,false);
        saveLocalStorage('CV',{link:linkName});
        this.props.dispatch(cv_setLink(linkName));
    }

    //обновляем ссылку в базе приложения
    updateLink = async (evt) => {
        let linkName = this.props.user;
        let stateToSave = {style:this.props.stylePage,blocks:this.props.blocks};
        evt.target.parentNode.classList.add('exiting');
        let timer = setTimeout(()=>{
                            this.setState({linkIsUpdated:true});
                            clearTimeout(timer);
                            return;}
                    ,500);
        saveFirebase('Links',linkName,stateToSave,false);
    }

    //открыть меню (мобил.версия)
    openMenuMobile = () => {
        this.menu.classList.toggle('header__menu--show');
    }
    
    render () {
        let cvBlocksCode = this.props.blocks.map( b => {
            return <CvBlock key={b.id} id={b.id} data={b} editable={false}></CvBlock>
        });

        let linkCode;
        if (this.props.link) {
            linkCode = <React.Fragment>
                            {!this.state.linkIsUpdated &&
                                <li className='header__menu-item'>
                                    <button className='header__button header__button--link-update' onClick={this.updateLink}>Update link</button>
                                </li>
                            }
                            <li className='header__menu-item'>
                                <NavLink to={'/' + this.props.link} className='header__button header__button--show' target="_blank">Open link ({this.props.link})</NavLink>
                            </li>
                        </React.Fragment>
        } else {
            linkCode = <li className='header__menu-item'>
                            <button className='header__button header__button--link-create' onClick={this.createLink}>Create Link</button>
                        </li>
        }

        return (
            <React.Fragment>
                {!this.state.viewForPrint && (
                    <header className={'header ' + this.props.transitionClass}>
                        <ul className='header__menu' ref={(f) => this.menu = f}>
                            <Media query="(min-width: 768px)">
                                <li className='header__menu-item'>
                                    <button className='header__button header__button--print' onClick={this.viewForPrint}>View for print</button>
                                </li>
                            </Media>
                            <li className='header__menu-item'>
                                <NavLink to='/' className='header__button header__button--edit'>Back to edit</NavLink>    
                            </li>
                            {linkCode}
                        </ul>
                        <Media query="(max-width: 767px)">
                            <button className='header__button header__button--menu' onClick={this.openMenuMobile}/>
                        </Media>
                    </header>
                )}
                <main className={'main ' + this.props.transitionClass}>
                    {this.state.viewForPrint && (
                        <div className='header__print no-print'>
                            <span>You can save this page to PDF-file. Open a webpage in Chrome, press Ctrl+P to open the Print dialog and change the destination printer to “Save as PDF”. Hit the “Print” button and the webpage will download as a PDF document
                            </span>
                            <NavLink to='/' className='header__button header__button--edit'>Back to edit</NavLink>
                        </div>
                    )}
                    <div className={'cv-view' + (this.state.viewForPrint?' cv-view-print':'')}>
                        <div className='cv' style={createStyle(this.props.stylePage)}>
                            {cvBlocksCode}
                        </div>
                    </div>
                </main>
            </React.Fragment>
        );
    }
}

const mapStateToProps = function (state) {
    return {
        blocks: state.cvData.blocks,
        stylePage: state.cvData.stylePage,
        user: state.cvData.user,
        link: state.cvData.link,
    };
};

export default connect(mapStateToProps)(CvView);
