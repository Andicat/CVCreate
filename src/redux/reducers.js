import { combineReducers } from 'redux';

import cvDataReducer from "./cvDataReducer";

let combinedReducer=combineReducers({
    cvData: cvDataReducer,
});

export default combinedReducer;
