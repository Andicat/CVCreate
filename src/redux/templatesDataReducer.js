import { TEMPLATE_LOAD } from './templatesDataAC';

const initState = {
    templatesArr: [],
    imageUrl: null,
}

function templatesDataReducer(state = initState, action) {

    switch (action.type) {
        //load
        case TEMPLATE_LOAD: {
            let newState = {templatesArr:action.data.templates, imageUrl:action.data.image};
            return newState;
        }
        default:
            return state;
    }
}

export default templatesDataReducer;
