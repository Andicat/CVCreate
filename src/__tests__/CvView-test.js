"use strict";

import {renderComponentWithStore, storeEmpty, storeWithData, storeWithLink} from './utils-test';
import {cv_setLink} from '../redux/cvDataAC';

import CvView from '../components/CvView';

let component;

describe('CvView render', () => {

    it('Рендер без данных', () => {
        component = renderComponentWithStore(CvView,storeEmpty);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер с данными', () => {
        component = renderComponentWithStore(CvView,storeWithData);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер с уже существующей ссылкой', () => {
        component = renderComponentWithStore(CvView,storeWithLink);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер для печати', () => {
        component = renderComponentWithStore(CvView,storeWithData);
        let buttonPrint = component.root.find( el => el.props.className=='header__button header__button--print');
        buttonPrint.props.onClick();
        expect(component.toJSON()).toMatchSnapshot();
    });
});

describe('CvView actions', () => {

    it('createLink', () => {
        component = renderComponentWithStore(CvView,storeWithData);
        let buttonLinkCreate = component.root.find( el => el.props.className=='header__button header__button--link-create');
        buttonLinkCreate.props.onClick();
        const expectedActions = [
            cv_setLink(storeWithData.getState().cvData.user),
        ];
        expect(storeWithData.getActions()).toEqual(expectedActions);
    });
});
