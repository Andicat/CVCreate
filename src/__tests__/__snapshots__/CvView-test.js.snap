// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvView render Рендер без данных 1`] = `
Array [
  <header
    className="header undefined"
  >
    <ul
      className="header__menu"
    >
      <li
        className="header__menu-item"
      >
        <button
          className="header__button header__button--print"
          onClick={[Function]}
        >
          View for print
        </button>
      </li>
      <li
        className="header__menu-item"
      >
        <a
          aria-current="page"
          className="header__button header__button--edit active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Back to edit
        </a>
      </li>
      <li
        className="header__menu-item"
      >
        <button
          className="header__button header__button--link-create"
          onClick={[Function]}
        >
          Create Link
        </button>
      </li>
    </ul>
    <button
      className="header__button header__button--menu"
      onClick={[Function]}
    />
  </header>,
  <main
    className="main undefined"
  >
    <div
      className="cv-view"
    >
      <div
        className="cv"
        style={
          Object {
            "backgroundColor": "#ffffff",
          }
        }
      />
    </div>
  </main>,
]
`;

exports[`CvView render Рендер для печати 1`] = `
<main
  className="main undefined"
>
  <div
    className="header__print no-print"
  >
    <span>
      You can save this page to PDF-file. Open a webpage in Chrome, press Ctrl+P to open the Print dialog and change the destination printer to “Save as PDF”. Hit the “Print” button and the webpage will download as a PDF document
    </span>
    <a
      aria-current="page"
      className="header__button header__button--edit active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Back to edit
    </a>
  </div>
  <div
    className="cv-view"
  >
    <div
      className="cv"
      style={
        Object {
          "backgroundColor": "#ffffff",
        }
      }
    >
      <div
        className="cv__block entered"
        onClick={[Function]}
        style={
          Object {
            "height": "66px",
            "left": "290.08598862935975px",
            "top": "52.08598862935976px",
            "width": "149px",
          }
        }
      >
        <span
          className="cv__element cv__element--text"
          contentEditable={false}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#1bac38",
              "fontFamily": "Roboto",
              "fontSize": "40px",
              "fontStyle": "normal",
              "fontWeight": "bold",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={false}
        >
          Big text
        </span>
      </div>
      <div
        className="cv__block entered"
        onClick={[Function]}
        style={
          Object {
            "height": "142px",
            "left": "60.43650934333691px",
            "top": "44.43650934333691px",
            "width": "182px",
          }
        }
      >
        <img
          alt=""
          className="cv__element cv__element--image"
          data-elem={true}
          onClick={[Function]}
          style={
            Object {
              "file": "",
              "opacity": 1,
            }
          }
        />
      </div>
      <div
        className="cv__block entered"
        onClick={[Function]}
        style={
          Object {
            "height": "62px",
            "left": "303.173880178947px",
            "top": "142.173880178947px",
            "width": "119px",
          }
        }
      >
        <div
          className="cv__element cv__element--group"
          style={Object {}}
        >
          <span
            className="cv__element cv__element--text"
            contentEditable={false}
            data-elem={true}
            onBlur={[Function]}
            onClick={[Function]}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Roboto",
                "fontSize": "20px",
                "fontStyle": "normal",
                "fontWeight": "bold",
                "padding": "0px 0px 0px 1px",
                "textAlign": "start",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
            suppressContentEditableWarning={false}
          >
            Your header
          </span>
          <span
            className="cv__element cv__element--text"
            contentEditable={false}
            data-elem={true}
            onBlur={[Function]}
            onClick={[Function]}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Roboto",
                "fontSize": "16px",
                "fontStyle": "normal",
                "fontWeight": "normal",
                "padding": "0px 0px 0px 1px",
                "textAlign": "start",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
            suppressContentEditableWarning={false}
          >
            your text
          </span>
        </div>
      </div>
      <div
        className="cv__block entered"
        onClick={[Function]}
        style={
          Object {
            "height": "50px",
            "left": "72.88890908532193px",
            "top": "248.88890908532193px",
            "width": "179px",
          }
        }
      >
        <div
          className="cv__element cv__element--dots-row"
          data-elem={true}
          onClick={[Function]}
          style={
            Object {
              "addcolor": "#E6E6E6",
              "addcount": 3,
              "maincolor": "#E05B49",
              "maincount": 5,
              "radius": 10,
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "#E05B49",
                "borderRadius": "50%",
                "height": "10px",
                "margin": "0 5px",
                "width": "10px",
              }
            }
          />
          <div
            style={
              Object {
                "backgroundColor": "#E05B49",
                "borderRadius": "50%",
                "height": "10px",
                "margin": "0 5px",
                "width": "10px",
              }
            }
          />
          <div
            style={
              Object {
                "backgroundColor": "#E05B49",
                "borderRadius": "50%",
                "height": "10px",
                "margin": "0 5px",
                "width": "10px",
              }
            }
          />
          <div
            style={
              Object {
                "backgroundColor": "#E05B49",
                "borderRadius": "50%",
                "height": "10px",
                "margin": "0 5px",
                "width": "10px",
              }
            }
          />
          <div
            style={
              Object {
                "backgroundColor": "#E05B49",
                "borderRadius": "50%",
                "height": "10px",
                "margin": "0 5px",
                "width": "10px",
              }
            }
          />
          <div
            style={
              Object {
                "backgroundColor": "#E6E6E6",
                "borderRadius": "50%",
                "height": "10px",
                "margin": "0 5px",
                "width": "10px",
              }
            }
          />
          <div
            style={
              Object {
                "backgroundColor": "#E6E6E6",
                "borderRadius": "50%",
                "height": "10px",
                "margin": "0 5px",
                "width": "10px",
              }
            }
          />
          <div
            style={
              Object {
                "backgroundColor": "#E6E6E6",
                "borderRadius": "50%",
                "height": "10px",
                "margin": "0 5px",
                "width": "10px",
              }
            }
          />
        </div>
      </div>
      <div
        className="cv__block entered"
        onClick={[Function]}
        style={
          Object {
            "height": "161px",
            "left": "322.66776857728195px",
            "top": "238.66776857728195px",
            "width": "171px",
          }
        }
      >
        <div
          className="cv__element cv__element--figure"
          data-elem={true}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#6AABB5",
              "borderRadius": "50%",
              "opacity": 1,
            }
          }
        />
      </div>
      <div
        className="cv__block entered"
        onClick={[Function]}
        style={
          Object {
            "height": "111px",
            "left": "487.9319841589094px",
            "top": "92.93198415890942px",
            "width": "252px",
          }
        }
      >
        <div
          className="cv__element cv__element--group"
          style={Object {}}
        >
          <span
            className="cv__element cv__element--text"
            contentEditable={false}
            data-elem={true}
            onBlur={[Function]}
            onClick={[Function]}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Roboto",
                "fontSize": "18px",
                "fontStyle": "normal",
                "fontWeight": "bold",
                "padding": "0px 0px 0px 1px",
                "textAlign": "start",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
            suppressContentEditableWarning={false}
          >
            Your position
          </span>
          <span
            className="cv__element cv__element--text"
            contentEditable={false}
            data-elem={true}
            onBlur={[Function]}
            onClick={[Function]}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Roboto",
                "fontSize": "18px",
                "fontStyle": "normal",
                "fontWeight": "normal",
                "padding": "0px 0px 0px 1px",
                "textAlign": "start",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
            suppressContentEditableWarning={false}
          >
            Company
          </span>
          <span
            className="cv__element cv__element--text"
            contentEditable={false}
            data-elem={true}
            onBlur={[Function]}
            onClick={[Function]}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Roboto",
                "fontSize": "16px",
                "fontStyle": "italic",
                "fontWeight": "normal",
                "padding": "0px 0px 0px 1px",
                "textAlign": "start",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
            suppressContentEditableWarning={false}
          >
            period
          </span>
          <span
            className="cv__element cv__element--text"
            contentEditable={false}
            data-elem={true}
            onBlur={[Function]}
            onClick={[Function]}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Roboto",
                "fontSize": "16px",
                "fontStyle": "normal",
                "fontWeight": "normal",
                "padding": "0px 0px 0px 1px",
                "textAlign": "start",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
            suppressContentEditableWarning={false}
          >
            your competencies and results
          </span>
        </div>
      </div>
    </div>
  </div>
</main>
`;

exports[`CvView render Рендер с данными 1`] = `
Array [
  <header
    className="header undefined"
  >
    <ul
      className="header__menu"
    >
      <li
        className="header__menu-item"
      >
        <button
          className="header__button header__button--print"
          onClick={[Function]}
        >
          View for print
        </button>
      </li>
      <li
        className="header__menu-item"
      >
        <a
          aria-current="page"
          className="header__button header__button--edit active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Back to edit
        </a>
      </li>
      <li
        className="header__menu-item"
      >
        <button
          className="header__button header__button--link-create"
          onClick={[Function]}
        >
          Create Link
        </button>
      </li>
    </ul>
    <button
      className="header__button header__button--menu"
      onClick={[Function]}
    />
  </header>,
  <main
    className="main undefined"
  >
    <div
      className="cv-view"
    >
      <div
        className="cv"
        style={
          Object {
            "backgroundColor": "#ffffff",
          }
        }
      >
        <div
          className="cv__block entered"
          onClick={[Function]}
          style={
            Object {
              "height": "66px",
              "left": "290.08598862935975px",
              "top": "52.08598862935976px",
              "width": "149px",
            }
          }
        >
          <span
            className="cv__element cv__element--text"
            contentEditable={false}
            data-elem={true}
            onBlur={[Function]}
            onClick={[Function]}
            style={
              Object {
                "color": "#1bac38",
                "fontFamily": "Roboto",
                "fontSize": "40px",
                "fontStyle": "normal",
                "fontWeight": "bold",
                "padding": "0px 0px 0px 1px",
                "textAlign": "start",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
            suppressContentEditableWarning={false}
          >
            Big text
          </span>
        </div>
        <div
          className="cv__block entered"
          onClick={[Function]}
          style={
            Object {
              "height": "142px",
              "left": "60.43650934333691px",
              "top": "44.43650934333691px",
              "width": "182px",
            }
          }
        >
          <img
            alt=""
            className="cv__element cv__element--image"
            data-elem={true}
            onClick={[Function]}
            style={
              Object {
                "file": "",
                "opacity": 1,
              }
            }
          />
        </div>
        <div
          className="cv__block entered"
          onClick={[Function]}
          style={
            Object {
              "height": "62px",
              "left": "303.173880178947px",
              "top": "142.173880178947px",
              "width": "119px",
            }
          }
        >
          <div
            className="cv__element cv__element--group"
            style={Object {}}
          >
            <span
              className="cv__element cv__element--text"
              contentEditable={false}
              data-elem={true}
              onBlur={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Roboto",
                  "fontSize": "20px",
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "padding": "0px 0px 0px 1px",
                  "textAlign": "start",
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
              suppressContentEditableWarning={false}
            >
              Your header
            </span>
            <span
              className="cv__element cv__element--text"
              contentEditable={false}
              data-elem={true}
              onBlur={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Roboto",
                  "fontSize": "16px",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "padding": "0px 0px 0px 1px",
                  "textAlign": "start",
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
              suppressContentEditableWarning={false}
            >
              your text
            </span>
          </div>
        </div>
        <div
          className="cv__block entered"
          onClick={[Function]}
          style={
            Object {
              "height": "50px",
              "left": "72.88890908532193px",
              "top": "248.88890908532193px",
              "width": "179px",
            }
          }
        >
          <div
            className="cv__element cv__element--dots-row"
            data-elem={true}
            onClick={[Function]}
            style={
              Object {
                "addcolor": "#E6E6E6",
                "addcount": 3,
                "maincolor": "#E05B49",
                "maincount": 5,
                "radius": 10,
              }
            }
          >
            <div
              style={
                Object {
                  "backgroundColor": "#E05B49",
                  "borderRadius": "50%",
                  "height": "10px",
                  "margin": "0 5px",
                  "width": "10px",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundColor": "#E05B49",
                  "borderRadius": "50%",
                  "height": "10px",
                  "margin": "0 5px",
                  "width": "10px",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundColor": "#E05B49",
                  "borderRadius": "50%",
                  "height": "10px",
                  "margin": "0 5px",
                  "width": "10px",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundColor": "#E05B49",
                  "borderRadius": "50%",
                  "height": "10px",
                  "margin": "0 5px",
                  "width": "10px",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundColor": "#E05B49",
                  "borderRadius": "50%",
                  "height": "10px",
                  "margin": "0 5px",
                  "width": "10px",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundColor": "#E6E6E6",
                  "borderRadius": "50%",
                  "height": "10px",
                  "margin": "0 5px",
                  "width": "10px",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundColor": "#E6E6E6",
                  "borderRadius": "50%",
                  "height": "10px",
                  "margin": "0 5px",
                  "width": "10px",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundColor": "#E6E6E6",
                  "borderRadius": "50%",
                  "height": "10px",
                  "margin": "0 5px",
                  "width": "10px",
                }
              }
            />
          </div>
        </div>
        <div
          className="cv__block entered"
          onClick={[Function]}
          style={
            Object {
              "height": "161px",
              "left": "322.66776857728195px",
              "top": "238.66776857728195px",
              "width": "171px",
            }
          }
        >
          <div
            className="cv__element cv__element--figure"
            data-elem={true}
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#6AABB5",
                "borderRadius": "50%",
                "opacity": 1,
              }
            }
          />
        </div>
        <div
          className="cv__block entered"
          onClick={[Function]}
          style={
            Object {
              "height": "111px",
              "left": "487.9319841589094px",
              "top": "92.93198415890942px",
              "width": "252px",
            }
          }
        >
          <div
            className="cv__element cv__element--group"
            style={Object {}}
          >
            <span
              className="cv__element cv__element--text"
              contentEditable={false}
              data-elem={true}
              onBlur={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Roboto",
                  "fontSize": "18px",
                  "fontStyle": "normal",
                  "fontWeight": "bold",
                  "padding": "0px 0px 0px 1px",
                  "textAlign": "start",
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
              suppressContentEditableWarning={false}
            >
              Your position
            </span>
            <span
              className="cv__element cv__element--text"
              contentEditable={false}
              data-elem={true}
              onBlur={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Roboto",
                  "fontSize": "18px",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "padding": "0px 0px 0px 1px",
                  "textAlign": "start",
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
              suppressContentEditableWarning={false}
            >
              Company
            </span>
            <span
              className="cv__element cv__element--text"
              contentEditable={false}
              data-elem={true}
              onBlur={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Roboto",
                  "fontSize": "16px",
                  "fontStyle": "italic",
                  "fontWeight": "normal",
                  "padding": "0px 0px 0px 1px",
                  "textAlign": "start",
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
              suppressContentEditableWarning={false}
            >
              period
            </span>
            <span
              className="cv__element cv__element--text"
              contentEditable={false}
              data-elem={true}
              onBlur={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Roboto",
                  "fontSize": "16px",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "padding": "0px 0px 0px 1px",
                  "textAlign": "start",
                  "textDecoration": "none",
                  "textTransform": "none",
                }
              }
              suppressContentEditableWarning={false}
            >
              your competencies and results
            </span>
          </div>
        </div>
      </div>
    </div>
  </main>,
]
`;

exports[`CvView render Рендер с уже существующей ссылкой 1`] = `
Array [
  <header
    className="header undefined"
  >
    <ul
      className="header__menu"
    >
      <li
        className="header__menu-item"
      >
        <button
          className="header__button header__button--print"
          onClick={[Function]}
        >
          View for print
        </button>
      </li>
      <li
        className="header__menu-item"
      >
        <a
          aria-current="page"
          className="header__button header__button--edit active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Back to edit
        </a>
      </li>
      <li
        className="header__menu-item"
      >
        <button
          className="header__button header__button--link-update"
          onClick={[Function]}
        >
          Update link
        </button>
      </li>
      <li
        className="header__menu-item"
      >
        <a
          aria-current={null}
          className="header__button header__button--show"
          href="/test"
          onClick={[Function]}
          target="_blank"
        >
          Open link (
          test
          )
        </a>
      </li>
    </ul>
    <button
      className="header__button header__button--menu"
      onClick={[Function]}
    />
  </header>,
  <main
    className="main undefined"
  >
    <div
      className="cv-view"
    >
      <div
        className="cv"
        style={
          Object {
            "backgroundColor": "#ffffff",
          }
        }
      >
        <div
          className="cv__block entered"
          onClick={[Function]}
          style={
            Object {
              "height": "66px",
              "left": "290.08598862935975px",
              "top": "52.08598862935976px",
              "width": "149px",
            }
          }
        >
          <span
            className="cv__element cv__element--text"
            contentEditable={false}
            data-elem={true}
            onBlur={[Function]}
            onClick={[Function]}
            style={
              Object {
                "color": "#1bac38",
                "fontFamily": "Roboto",
                "fontSize": "40px",
                "fontStyle": "normal",
                "fontWeight": "bold",
                "padding": "0px 0px 0px 1px",
                "textAlign": "start",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
            suppressContentEditableWarning={false}
          >
            Big text
          </span>
        </div>
      </div>
    </div>
  </main>,
]
`;
